### DISCUSSION TITLE
Improve bottom's CPU usage

### DISCUSSION THREAD

**Discussion by 12932** [ClementTsang/bottom#915] [General]
Today I ran a test between top/htop/bottom/btop to compare how much CPU they all use. The results SHOCKED me. I understand `top` is nothing like `bottom`. I'm just using it as a sort of baseline.

I ran them all for exactly **2 hours** with update rates set to **1 second** and measured their **total** CPU time on my 64-bit Linux (Debian) machine. bottom was installed with `cargo install` using rust stable.

| Software  | CPU time |
| ------------- | ------------- |
| top 3.3.17 | 29 seconds |
| htop 3.0.5  | 40 seconds |
| btop 1.2.13 | 64 seconds |
| bottom 0.6.8 | 183 seconds |

**Comment by ClementTsang** [comment]
Performance is something I'm always striving to improve, yeah. Curious how the current master branch looks.

**Reply by 12932** [reply]
Here's with bottom 0.6.9 (current master branch in --release mode ofcourse) in basic view. For this test btop was also in a "basic" view with only process list and CPU showing. These were tested for slightly less than 2 hours

| Software  | CPU time |
| ------------- | ------------- |
| top 3.3.17 | 26 seconds |
| htop 3.0.5  | 36 seconds |
| btop 1.2.13 | 48 seconds |
| bottom 0.6.9 | 153 seconds |

**Reply by ClementTsang** [reply]
Thanks, good to see there seems to be at least some improvements... definitely will look to keep improving it though. I know where it's generally spending the most CPU time from flamegraphs so hopefully tackling those will help a bit.
